### PR TITLE
Refs task #391947 #391944 [Payload history] [List & search] coding and unit test

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#391947](https://dev.sun-asterisk.com/issues/391947)
- [#391944](https://dev.sun-asterisk.com/issues/391944)

## What's this PR do ?

- [x] Coding & Unit test for list and search payload history
- [x] Fix bugs on function B

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

- All record

image

- By webhook
image

- By both webhook & status 
image

- search not found
image

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

File: app/Repositories/Eloquents/PayloadHistoryRepository.php [hunk 1]
```diff
@@ -16,4 +18,20 @@ public function find($id)
...
+            $searchParams = $this->handleSearchParams(['webhook', 'status'], $searchParams);
+
+            return $query->search($searchParams, $perPage);
+        } else {
+            return $query->paginate($perPage);
+        }
+    }
+
+
 }
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PayloadHistoryRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Classes.ClassDeclaration.CloseBraceAfterBody</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PayloadHistoryRepository.php</br>**Line**: 37</br>The closing brace for the class must go on the next line after the body

File: app/Http/Controllers/PayloadHistoryController.php [hunk 2]
```diff
@@ -3,30 +3,41 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use App\Enums\PayloadHistoryStatus;
 use App\Repositories\Interfaces\PayloadHistoryRepositoryInterface as PayloadHistoryRepository;
 use App\Repositories\Interfaces\MessageHistoryRepositoryInterface as MessageHistoryRepository;
-
+use App\Repositories\Interfaces\WebhookRepositoryInterface as WebhookRepository;
+use Auth;
```
Comment (framgiaciv3) [inline app/Http/Controllers/PayloadHistoryController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse</br>**Severity**: error</br>**File**: app/Http/Controllers/PayloadHistoryController.php</br>**Line**: 10</br>There must be one blank line after the last USE statement; 0 found;

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/PayloadHistoryController.php [hunk 3]
```diff
@@ -3,30 +3,42 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use App\Enums\PayloadHistoryStatus;
 use App\Repositories\Interfaces\PayloadHistoryRepositoryInterface as PayloadHistoryRepository;
 use App\Repositories\Interfaces\MessageHistoryRepositoryInterface as MessageHistoryRepository;
+use App\Repositories\Interfaces\WebhookRepositoryInterface as WebhookRepository;
+use Auth;
```
Comment (nguyenthinhhn) [inline app/Http/Controllers/PayloadHistoryController.php]: Auth không dùng thì bỏ đi a

File: routes/web.php [hunk 4]
```diff
@@ -45,4 +45,5 @@
         'index'
     ]);
     Route::resource('webhooks.payloads', 'PayloadController')->except('show');
+    Route::resource('payload_histories', 'PayloadHistoryController')->only('index');
```
Comment (quanghung97) [inline routes/web.php]: trên có route history rồi sao ko dùng chung đi nhỉ

Comment (quanghung97) [reply routes/web.php]: với lại nếu dùng riêng url ko được gạch dưới

**Review by quanghung97 — APPROVED** [review]
OK

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Review by nguyenthinhhn — APPROVED** [review]

File: app/Http/Controllers/PayloadHistoryController.php (hunk 3 above)
Comment (ghost) [reply app/Http/Controllers/PayloadHistoryController.php]: oke e anh có debug chỗ này quên xóa mất :D

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**